### PR TITLE
CLOUDP-200720: UX improvements

### PR DESCRIPTION
**Pull request by blva**
## Proposed changes

_Jira ticket:_[CLOUDP-200720](https://jira.mongodb.org/browse/CLOUDP-200720)

1. fix atlas setup --type LOCAL
2. fix index duplication creation to work the same way as atlas api
## Checklist

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/mongodbclient/search_index.go [hunk 1]
```diff
@@ -128,6 +118,21 @@ func (o *database) SearchIndex(ctx context.Context, id string) (*admin.ClusterSe
...
+	indexes, err := o.SearchIndexes(ctx, collection)
+	if err != nil {
+		return nil, err
+	}
+
+	for index := range indexes {
+		if indexes[index].Name == name && indexes[index].Database == o.db.Name() {
+			return indexes[index], nil
+		}
+	}
```
Comment (tibulca) [inline internal/mongodbclient/search_index.go:131]:
```suggestion
	for _, index := range indexes {
		if index.Name == name && index.Database == o.db.Name() {
			return index, nil
		}
	}
```

File: internal/cli/atlas/deployments/search/indexes/create.go [hunk 2]
```diff
@@ -78,7 +81,12 @@ func (opts *CreateOpts) Run(ctx context.Context) error {
 		return err
 	}
 
-	opts.index, err = opts.mongodbClient.Database(opts.index.Database).CreateSearchIndex(ctx, opts.index.CollectionName, opts.index)
+	db := opts.mongodbClient.Database(opts.index.Database)
+	if _, err = db.SearchIndexByName(ctx, opts.index.Name, opts.index.CollectionName); err != mongodbclient.ErrSearchIndexNotFound {
+		return ErrSearchIndexDuplicated
+	}
```
Comment (tibulca) [inline internal/cli/atlas/deployments/search/indexes/create.go:87]: should this be something like this?
```suggestion
	if idx, _ = db.SearchIndexByName(ctx, opts.index.Name, opts.index.CollectionName); idx != nil {
		return ErrSearchIndexAlreadyExists
	}
```

File: internal/cli/atlas/deployments/search/indexes/create.go [hunk 3]
```diff
@@ -38,10 +39,12 @@ import (
 const (
 	namePattern        = "^[a-zA-Z0-9][a-zA-Z0-9-]*$"
 	connectWaitSeconds = 10
-	createTemplate     = "Search index created with ID {{.IndexID}}\n"
+	createTemplate     = "Search index created with ID: {{.IndexID}}\n"
 	notFoundState      = "NOT_FOUND"
 )
 
+var ErrSearchIndexDuplicated = errors.New("search index is duplicated")
```
Comment (tibulca) [inline internal/cli/atlas/deployments/search/indexes/create.go:46]: [optional] wdyt about:
```suggestion
var ErrSearchIndexAlreadyExists = errors.New("search index already exists")
```

Comment (blva) [reply internal/cli/atlas/deployments/search/indexes/create.go:46]: based myself on index API output, no strong opinion though, wdyt? 
```shell

Error: [https://cloud.mongodb.com/api/atlas/v2/groups/623202922f697375cac9414a/clusters/Cluster07293/fts/indexes](https://www.google.com/url?q=https://cloud.mongodb.com/api/atlas/v2/groups/623202922f697375cac9414a/clusters/Cluster07293/fts/indexes&sa=D&source=docs&ust=1695219566833407&usg=AOvVaw36Yeiu-SptRWBb-WkOaAH4) POST: HTTP 400 (Error code: "ATLAS_FTS_DUPLICATE_INDEX") Detail: Duplicate Index. Reason: Bad Request. Params: []
``
`

File: internal/cli/atlas/deployments/search/indexes/create.go (hunk 2 above)
Comment (blva) [reply internal/cli/atlas/deployments/search/indexes/create.go:87]: sure, that works as well

File: internal/cli/atlas/deployments/search/indexes/create.go [hunk 4]
```diff
@@ -78,7 +81,12 @@ func (opts *CreateOpts) Run(ctx context.Context) error {
...
-	opts.index, err = opts.mongodbClient.Database(opts.index.Database).CreateSearchIndex(ctx, opts.index.CollectionName, opts.index)
+	db := opts.mongodbClient.Database(opts.index.Database)
+	if idx, _ := db.SearchIndexByName(ctx, opts.index.Name, opts.index.CollectionName); idx != nil {
+		return ErrSearchIndexDuplicated
+	}
+
+	opts.index, err = db.CreateSearchIndex(ctx, opts.index.CollectionName, opts.index)
 	if err != nil {
 		return err
 	}
```
Comment (tibulca) [inline internal/cli/atlas/deployments/search/indexes/create.go:92]: [optional]
```
	if err != nil {
		return err
	}
	return nil
```
replace it with `return err`

File: internal/cli/atlas/deployments/setup.go [hunk 5]
```diff
@@ -782,7 +782,7 @@ func (opts *SetupOpts) Run(ctx context.Context) error {
 	}
 
 	telemetry.AppendOption(telemetry.WithDeploymentType(opts.DeploymentType))
-	if opts.DeploymentType == localCluster {
+	if strings.EqualFold(localCluster, opts.DeploymentType) {
```
Comment (tibulca) [inline internal/cli/atlas/deployments/setup.go:785]: thank you

**Review by tibulca — APPROVED** [review]

File: internal/cli/atlas/deployments/search/indexes/create.go (hunk 4 above)
Comment (blva) [reply internal/cli/atlas/deployments/search/indexes/create.go:92]: will do that in my next PR

**Pull request merged**